### PR TITLE
Ensure 'not exist' errors wrap os.ErrNotExist

### PR DESCRIPTION
**Pull request by campoy**
Updates #3873 

This PR ensures that all "not exist" types return an error for which errors.Is(err, os.ErrNotExist) is true.

### CODE REVIEW

**Review by natasha41575 — COMMENTED** [review]
The presubmit check is failing because the api module is missing a go.sum entry, you can see the error output here: https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_kustomize/3982/kustomize-presubmit-master/1403066536456884224

Besides that, the change looks okay to me, though could you clarify what benefit this provides?

**Comment by campoy** [comment]
Of course, yes.

I'm working on #3873 and I want to ensure that the behavior of the different implementations of `FileSystem` behave the same way.

Right now, some of the checks that I would do are failing because the error returned when opening a non existent file on FsOnDisk do wrap os.ErrNotExist, but not FsInMemory.

I also just sent an updated go.sum.

**Comment by campoy** [comment]
Regarding the multiple commits, I'm totally fine squashing them but I don't think I can request that myself.

Should I squash it manually on my end?

**Review by natasha41575 — COMMENTED** [review]
Minor nits

File: api/filesys/fsnode.go [hunk 1]
```diff
@@ -351,6 +352,9 @@ func (n *fsNode) IsDir(path string) bool {
 
 // ReadDir implements FileSystem.
 func (n *fsNode) ReadDir(path string) ([]string, error) {
+	if !n.Exists(path) {
+		return nil, fmt.Errorf("%s does not exist: %w", path, os.ErrNotExist)
```
[the remark this answers is not in the document]
Comment (campoy) [reply api/filesys/fsnode.go]: Fixed

File: api/filesys/fsnode.go [hunk 2]
```diff
@@ -236,7 +236,7 @@ func (n *fsNode) CleanedAbs(path string) (ConfirmedDir, string, error) {
 		return "", "", errors.Wrap(err, "unable to clean")
 	}
 	if node == nil {
-		return "", "", fmt.Errorf("'%s' doesn't exist", path)
+		return "", "", fmt.Errorf("'%s' doesn't exist: %w", path, os.ErrNotExist)
```
[the remark this answers is not in the document]
Comment (campoy) [reply api/filesys/fsnode.go]: Fixed

**Comment by natasha41575** [comment]
note: You can run `make prow-presubmit-check` locally to run this check. (The helm tests fail on my machine so I usually comment them out.)

**Comment by campoy** [comment]
After having to fix too many tests, I realized that there was a way to avoid having to change anything on the client's side by using a custom error type!

**Pull request merged**